maestro/docs/README_openai.md

```diff
@@ -0,0 +1,201 @@
+* **Max Tokens (Optional):** Set `MAESTRO_OPENAI_MAX_TOKENS` to a positive integer to limit the maximum number of tokens generated by the model.
+  * Example: `export MAESTRO_OPENAI_MAX_TOKENS=64000`
+* To enable **Open Telemetry** capture of LLM calls, set `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` for example `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT=http://localhost:4318/v1/traces`
+* **Extra Headers (Optional):** Set `MAESTRO_OPENAI_EXTRA_HEADERS` to a JSON string representing a dictionary of custom HTTP headers to send with requests to the OpenAI API or compatible endpoint. These are added via the `ModelSettings`.
+  * Example: `export MAESTRO_OPENAI_EXTRA_HEADERS='{"SECRET_ACCESS_KEY": "aB3dE5fG7h", "AI-Resource-Group": "ishaan-resource"}'`. **Note:** For security, the *values* of these headers will be obfuscated (shown as `*****`) when printed in the agent's startup logs, but the actual values will be sent to the API.
+  * Note: Ensure the JSON string is properly quoted for your shell environment. The example shows setting `SECRET_ACCESS_KEY` to a random 10-character alphanumeric value.
+* **Use LiteLLM Backend (Optional):** Set `MAESTRO_OPENAI_USE_LITELLM=true` to use the LiteLLM library as the backend instead of the default OpenAI client. This allows connecting to a wider range of LLM providers
+  * When enabled, `spec.model` in the agent definition should be the LiteLLM model string (e.g., `anthropic/claude-3-opus-20240229`, `groq/llama3-70b-8192`). See the LiteLLM Providers Docs for supported models.
+  * `OPENAI_API_KEY` will be used as the `api_key` passed to LiteLLM. Ensure this key is appropriate for the selected LiteLLM model provider.
+  * `OPENAI_BASE_URL` can be used to specify a custom LiteLLM proxy endpoint or a provider-specific base URL if needed.
+  * 'MAESTRO_OPENAI_EXTRA_HEADERS' will still be passed as above
+
+### Basic Agent Definition
+
+Define an agent with `spec.framework: openai` and `spec.mode: local` in your agent YAML file. The `spec.model: gpt-4o-mini` specifies which LLM to use.
+
+```yaml
+# Example: tests/agents/openai_agent/agents.yaml
+apiVersion: maestro/v1alpha1
+kind: Agent
+metadata:
+  name: openai_test_mcp
+  labels:
+    app: testapp_mcp
+spec:
+  model: "gpt-4o-mini" # Specify the model to use
+  description: test
+  instructions: You are a helpful agent. Respond to the users question, making use of any required tools
+  framework: openai
+  mode: local # 'local' mode uses this Maestro agent implementation
+```
+
+It is important to set the `OPENAI_API_KEY` environment variable - see [OpenAI's website](https://platform.openai.com/account/api-keys).
+
+This will connect to OpenUI using the [new Responses API](https://platform.openai.com/docs/guides/responses-vs-chat-completions) at `https://api.openai.com/v1`
+
+The full example can be run with:
+
+```shell
+maestro run tests/agents/openai_agent/agents.yaml test/agents/openai_agent/workflow.yaml
```

Comment:
`test/agents/openai_agent/workflow.yaml` to `tests/agents/openai_agent/workflow.yaml`